### PR TITLE
Add constructors to facilitate unit testing

### LINKED ISSUE #94: StanMsgHandlerArgs internal constructor and no interface makes unit-testing impossible
StanMsgHandlerArgs currently has an internal constructor, and it is not implementing an interface.
This makes it impossible to create unit-tests on methods that in some way rely on an object of this type.

Would it be possible for it to be opened up, or for there to be some factory that can produce a "default" version?

Edit: This also applies to StanMsg

### PR DESCRIPTION
**Pull request by ColinSullivan1**
Resolves #94.

I debated adding these to the connection factories, I think constructors are simpler and the path of least surprise.  If we do want a factory for these, a new factory type for testing might be the way to go.

Signed-off-by: Colin Sullivan <colin@synadia.com>

### CODE REVIEW

File: STAN.CLIENT/StanMsg.cs [hunk 1]
```diff
@@ -22,6 +22,28 @@ public sealed class StanMsg
...
+        /// </summary>
+        /// <remarks>
+        /// Objects of this type are normally generated internally by the NATS streaming client.
+        /// This constructor has been provided to facilitate application unit testing.
+        /// </remarks>
+        /// <param name="data">The message payload.</param>
+        /// <param name="redelivered">True if the message may have been redelivered.</param>
+        /// <param name="subject">Subject of the message</param>
+        /// <param name="timestamp">Message timestamp, nanoseconds since epoch (1/1/1970)</param>
+        /// <param name="subscription">Subscription of the message.  May be null</param>
```
Comment (watfordgnf) [inline STAN.CLIENT/StanMsg.cs]: This should probably be noted that it needs to be an `AsyncSubscription`.

File: STAN.CLIENT/StanMsgHandlerArgs.cs [hunk 2]
```diff
@@ -21,7 +21,28 @@ public class StanMsgHandlerArgs : EventArgs
...
+        /// </summary>
+        /// <remarks>
+        /// Objects of this type are normally generated internally by the NATS streaming client.
+        /// This constructor has been provided to facilitate application unit testing.
+        /// </remarks>
+        /// <param name="data">The message payload.</param>
+        /// <param name="redelivered">True if the message may have been redelivered.</param>
+        /// <param name="subject">Subject of the message.</param>
+        /// <param name="timestamp">Message timestamp, nanoseconds since epoch.(1/1/1970)</param>
+        /// <param name="subscription">Subscription of the message.  May be null.</param>
```
Comment (watfordgnf) [inline STAN.CLIENT/StanMsgHandlerArgs.cs]: Same note as before, the cast in `StanMsg.ctor()` necessitates that this is an `AsyncSubscription`.

File: STAN.CLIENT/StanMsg.cs (hunk 1 above)
Comment (ColinSullivan1) [reply STAN.CLIENT/StanMsg.cs]: Good point...  I'll add it.  I also forgot sequence number.

**Comment by ColinSullivan1** [comment]
Regarding the internal `AsyncSubscription`, we need it...  a reference to the subscription is required to manually ack.  We could to add the manualAck method to the IStanSubscription interface and just use the interface internally, but imo that could create confusion in usage and would veer from the go client.

**Pull request merged**